File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const counts = {};
  for (let i = 0; i < nums.length; i++) {
    if (!(nums[i] in counts)) counts[nums[i]] = i;
    else {
      if (Math.abs(counts[nums[i]] - i) <= k) {
        return true;
      }
        if(nums[i] in counts){
        counts[nums[i]] = i;
      }
    }
  }
  return false;
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        map = {}
        for left in range(len(nums)):
            if nums[left] in map and abs( map[nums[left]] - left ) <= k:
                return True
            map[nums[left]] = left
            
        return False